<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Capital P Dangit"
    >
    <standard>
    <![CDATA[
The correct spelling of "WordPress" should be used in text strings, comments and object names.

In select cases, when part of an identifier or a URL, WordPress does not have to be capitalized.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: WordPress is correctly capitalized.">
        <![CDATA[
class <em>WordPress</em>_Example {

    /**
     * This function is about <em>WordPress</em>.
     */
    public function explain() {
        echo 'This is an explanation
            about <em>WordPress</em>.';
    }
}
        ]]>
        </code>
        <code title="Invalid: WordPress is not correctly capitalized.">
        <![CDATA[
class <em>Wordpress_Example</em> {

    /**
     * This function is about <em>Wordpress</em>.
     */
    public function explain() {
        echo 'This is an explanation
            about <em>wordpress</em>.';
    }
}
        ]]>
        </code>
    </code_comparison>
</documentation>
