<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="Deprecated Function Parameter Values"
    >
    <standard>
    <![CDATA[
    Please refrain from using deprecated WordPress function parameter values.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: passing a valid function parameter value.">
        <![CDATA[
bloginfo( <em>'url'</em> );
        ]]>
        </code>
        <code title="Invalid: passing a deprecated function parameter value.">
        <![CDATA[
bloginfo ( <em>'home'</em> ); // Deprecated WP 2.2.0.
        ]]>
        </code>
    </code_comparison>
</documentation>
