<?xml version="1.0"?>
<documentation xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:noNamespaceSchemaLocation="https://phpcsstandards.github.io/PHPCSDevTools/phpcsdocs.xsd"
    title="High Posts Per Page Limit"
    >
    <standard>
    <![CDATA[
Using "posts_per_page" or "numberposts" with the value set to an high number opens up the potential for making requests slow if the query ends up querying thousands of posts.

You should always fetch the lowest number possible that still gives you the number of results you find acceptable.
    ]]>
    </standard>
    <code_comparison>
        <code title="Valid: posts_per_page is not over limit (default 100).">
        <![CDATA[
$args = array(
    'posts_per_page' => <em>-1</em>,
);
$args = array(
    'posts_per_page' => <em>100</em>,
);
$args = array(
    'posts_per_page' => <em>'10'</em>,
);

$query_args['posts_per_page'] = <em>100</em>;

_query_posts( 'nopaging=1&posts_per_page=<em>50</em>' );
        ]]>
        </code>
        <code title="Invalid: posts_per_page is over limit (default 100).">
        <![CDATA[
$args = array(
    'posts_per_page' => <em>101</em>,
);

$query_args['posts_per_page'] = <em>200</em>;

_query_posts( 'nopaging=1&posts_per_page=<em>999</em>' );
        ]]>
        </code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: numberposts is not over limit (default 100).">
        <![CDATA[
$args = array(
    'numberposts' => <em>-1</em>,
);
$args = array(
    'numberposts' => <em>100</em>,
);
$args = array(
    'numberposts' => <em>'10'</em>,
);

$query_args['numberposts'] = <em>'-1'<em>;

_query_posts( 'numberposts=<em>50</em>' );
        ]]>
        </code>
        <code title="Invalid: numberposts is over limit (default 100).">
        <![CDATA[
$args = array(
    'numberposts' => <em>101</em>,
);

$query_args['numberposts'] = <em>'200'</em>;

_query_posts( 'numberposts=<em>999</em>' );
        ]]>
        </code>
    </code_comparison>
</documentation>
